### PR TITLE
feat(PLM-95): Provide simple find request method as a fallback method

### PR DESCRIPTION
**Pull request by kilikkuo**
[How]
1. Support find a request in _extract_request method
2. Add test case

[PLM-95]

Why need this change ? / Root Cause :
--------------

Changed Made :
--------------

Test Scope / Change Impact :
--------------

[PLM-95]: https://hardcoretech.atlassian.net/browse/PLM-95

### CODE REVIEW

File: data_spec_validator/__version__.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = '1.0.3'
+__version__ = '1.1.3'
```
Comment (xeonchen) [inline data_spec_validator/__version__.py:1]: 怎麼不是 `1.1.0`？

File: data_spec_validator/decorator/decorators.py [hunk 2]
```diff
@@ -75,6 +75,10 @@ def _extract_request(*args):
             req = args[1]
         return req
     else:
+        # Fallback to find the first request object
+        req = next(iter(filter(lambda o: _is_request(o), args)), None)
```
Comment (xeonchen) [inline data_spec_validator/decorator/decorators.py:79]: `filter` is already an iterator

**Review by xeonchen — APPROVED** [review]
r+ with comments

File: data_spec_validator/decorator/decorators.py (hunk 2 above)
Comment (kilikkuo) [reply data_spec_validator/decorator/decorators.py:79]: nice!

File: data_spec_validator/__version__.py (hunk 1 above)
Comment (kilikkuo) [reply data_spec_validator/__version__.py:1]: yeah, should make it 1.1.0, thanks

**Pull request merged**